extensions/eventtap/internal.m

```diff
@@ -173,12 +172,18 @@ static int eventtap_start(lua_State* L) {
                               eventtap_callback,
                               e);
 
-    CGEventTapEnable(e->tap, true);
-    e->runloopsrc = CFMachPortCreateRunLoopSource(NULL, e->tap, 0);
-    CFRunLoopAddSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
+    if (e->tap) {
+        e->running = true;
 
-    lua_settop(L,1);
-    return 1;
+        CGEventTapEnable(e->tap, true);
+        e->runloopsrc = CFMachPortCreateRunLoopSource(NULL, e->tap, 0);
+        CFRunLoopAddSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
+
+        lua_settop(L,1);
+        return 1;
+    } else {
+        showError(L, "Unable to create eventtap.  Is Accessibility enabled?");
+    }
```

Comment:
Sorry -- the examples I glanced at when trying to deciding wether to use showError or luaL_error didn't immediately follow showError with a return, and luaL_error *doesn't* return... perhaps the examples I perused fell through to an eventual return that I missed.  Returning nil would probably be appropriate.

I actually had a question sort of forming in my mind about that and hadn't decided if I should put it to the group yet or not... my current practice is that, except for specific getter/setter functions, class methods that I write that act on an object return that object... it allows things like `object:method1():method2():...:methodN()`

I noticed that back when I ported eventtap, I return nil for both start and stop if it is already in that state...  while I'm not sure that it is likely that eventtap would be chained in this manner, should we consider making sure all modules are consistent across the board in this respect? 

On the one hand, deciding yes would mean a more consistent experience for the end-user/local-developer... on the other hand, it would mean being more dogmatic about module requirements when  developers propose new core modules...  I'm not sure where I fall on this, but it seemed as good a time as any to bring it up.


> On Jun 16, 2015, at 3:41 AM, Chris Jones <notifications@github.com> wrote:
> 
> In extensions/eventtap/internal.m <https://github.com/Hammerspoon/hammerspoon/pull/318#discussion_r32501045>:
> 
> >  
> > -    lua_settop(L,1);
> > -    return 1;
> > +        CGEventTapEnable(e->tap, true);
> > +        e->runloopsrc = CFMachPortCreateRunLoopSource(NULL, e->tap, 0);
> > +        CFRunLoopAddSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
> > +
> > +        lua_settop(L,1);
> > +        return 1;
> > +    } else {
> > +        showError(L, "Unable to create eventtap.  Is Accessibility enabled?");
> > +    }
> This is missing a return statement. We should probably have the lua_settop(L, 1); return; here. This method is supposed to return the eventtap object.
> We could also arguably return nil and document that that happens in the event of an error.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Hammerspoon/hammerspoon/pull/318/files#r32501045>.
>